modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/streams/kinesis/source/KCLAction.scala

```diff
@@ -15,12 +15,40 @@ private sealed trait KCLAction
 
 private object KCLAction {
 
-  final case class ProcessRecords(shardId: String, processRecordsInput: ProcessRecordsInput) extends KCLAction
+  /**
+   * The action emitted by the ShardRecordProcessor when it receives new records
+   *
+   * @param await
+   *   A countdown latch used to backpressure the ShardRecordProcessor. The consumer of the queue
+   *   should release the countdown latch to unblock the ShardRecordProcessor and let it fetch more
+   *   records from Kinesis.
```

Comment:
Am I right that even tough we now have this `CountDownLatch` we don't wait for the `countDown()` before calling next `GetRecords`, we always have some records prefetched in advance, so there is no impact on throughput?

Reply:
By doing it this way, the 3 Sources (kinesis, pubsub, kafka) all behave roughly similarly in terms of how much pre-fetching they do.  I thought that was a nice benefit, that we can tune the downstream app for one source, and all other sources should behave roughly the same.

If a particular app finds it needs extra prefetching, then that app is free to call `.prefetch` on its source as the first step n the `Processing` class.  But so far there hasn't been an app that has needed that.